opentelemetry-proto/src/opentelemetry/proto/collector/logs/v1/logs_service_pb2_grpc.py

```diff
@@ -1,8 +1,38 @@
 # Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
 """Client and server classes corresponding to protobuf-defined services."""
 import grpc
+import warnings
 
-from opentelemetry.proto.collector.logs.v1 import logs_service_pb2 as opentelemetry_dot_proto_dot_collector_dot_logs_dot_v1_dot_logs__service__pb2
+from opentelemetry.proto.collector.logs.v1 import (
+    logs_service_pb2 as opentelemetry_dot_proto_dot_collector_dot_logs_dot_v1_dot_logs__service__pb2,
+)
+
+GRPC_GENERATED_VERSION = "1.65.5"
+GRPC_VERSION = grpc.__version__
+EXPECTED_ERROR_RELEASE = "1.66.0"
+SCHEDULED_RELEASE_DATE = "August 6, 2024"
+_version_not_supported = False
+
+try:
+    from grpc._utilities import first_version_is_lower
+
+    _version_not_supported = first_version_is_lower(
+        GRPC_VERSION, GRPC_GENERATED_VERSION
+    )
+except ImportError:
+    _version_not_supported = True
+
+if _version_not_supported:
+    warnings.warn(
```

Comment:
Any thoughts on how to protect users from this? Maybe we could use the oldest grpcio version supporting protobuf 5

Reply:
Nope. I did some tests locally using a different version for runtime and got the warnings during the tests 😓

Reply:
Maybe we can re-generate the code using version `1.63.1` instead

Reply:
Yes, that would probably work. Apologies if you already know this, but IIRC you have to check what version of protoc and the grpc plugin are included in the grpcio-tools pypi package version.

```console
$ python -m grpc_tools.protoc --version
libprotoc 26.1
```

And the version of libprotoc corresponds to the Python protobuf version in this table https://protobuf.dev/support/version-support/#python.

Reply:
Yeah I did that. From 1.63.0 it starts to use libprotoc 26.1